### PR TITLE
Ratings tests

### PR DESCRIPTION
**Pull request by AlexandraMoga**
These are a set of tests that are covering the Ratings card from the add-on detail page. They cover writing, editing, deleting reviews and some other related functionalities. These tests need to be run serially, since they are interdependent.

I need to change the CircleCI config to allow for parallel and serial test runs to work separately, due to an incompatibility between `pytest-xdist` and `pytest-depends` (which I need for the ratings tests, as some tests are interdependent). `pytest-xdist` will simply skip a test if its dependency has not run yet.

So far this has been a failure. I can't figure out why CircleCI remains stuck in the 'Run serial tests' step. It should exit and proceed to the next step once all the tests have been executed. @jrbenny35 maybe you can see something I don't in the config file? This is the output from my latest build run for the 'Run serial tests' step - https://circleci.com/api/v1.1/project/github/mozilla/addons-release-tests/654/output/105/0?file=true&allocation-id=610953e2c6973825de5c3092-0-build%2F4E73B2FE. There are no errors visible and pytest finishes the test execution. 
I've manually cancelled the job as it would not finish.

~~I'll also need to investigate further why some ratings tests are failing on CI, since I could not reproduce those errors locally.~~ Tests are passing now.

### CODE REVIEW

**Comment by AlexandraMoga** [comment]
Tests are passing with the latest version of `pytest-xdist`. @jrbenny35  I think this PR is ready for a proper review. Thanks for the help.

File: pages/desktop/details.py [hunk 1]
```diff
@@ -565,3 +599,105 @@ def ratings_card_summary(self):
...
+        def delete_rating_link(self):
+            # waits for the ratings stars to be fully loaded and editable
+            self.wait.until(
+                expected.invisibility_of_element_located(
+                    self._loaded_rating_stars_locator
+                )
+            )
+            return self.find_element(*self._delete_rating_link_locator)
+
+        def delete_confirm_button(self):
```
Comment (jrbenny35) [inline pages/desktop/details.py:633]: maybe this should have click in the name. This seems like it is named like a property not a function.
```suggestion
        def click_delete_confirm_button(self):
```

File: pages/desktop/details.py [hunk 2]
```diff
@@ -565,3 +599,105 @@ def ratings_card_summary(self):
...
+        def all_reviews_link(self):
+            self.find_element(*self._all_reviews_link_locator).click()
+            return Reviews(self.selenium, self.page.base_url).wait_for_page_to_load()
+
+        @property
+        def all_reviews_link_rating_count(self):
+            count = self.find_element(*self._all_reviews_link_locator).text
+            return int(count.split()[2])
+
+        def report_abuse(self):
```
Comment (jrbenny35) [inline pages/desktop/details.py:702]: This should also have click in the name.
```suggestion
        def click_report_abuse_button(self):
```

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
R+ just some nits

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,14 +20,23 @@ jobs:
           name: Install Firefox
           command: choco install firefox-nightly --pre
       - run:
-          name: Run tests
+          name: Run serial tests
```
[the remark this answers is not in the document]
Comment (AlexandraMoga) [reply .circleci/config.yml:22]: I've modified the circleci config to run the tests in parallel jobs. Please have a look and let me know what you think.

File: .circleci/config.yml [hunk 4]
```diff
@@ -20,18 +45,20 @@ jobs:
...
       - store_artifacts:
-          path: ui-test-release.html
+          path: parallel-test-results.html
+
 workflows:
   commit_workflow:
     jobs:
-      - test
+      - serial_tests
+      - parallel_tests
```
Comment (jrbenny35) [inline .circleci/config.yml:61]: WOOOOOOOOOOOOOOO

**Review by jrbenny35 — APPROVED** [review]

**Pull request merged**